platform/ext/common/gcc/tfm_common_s.ld

```diff
@@ -589,6 +598,32 @@ SECTIONS
 
 #endif /* TFM_LVL != 1 */
 
+#ifndef TFM_MULTI_CORE_TOPOLOGY
+    /*
+     * Place the CMSE Veneers (containing the SG instruction) after the code, in a
+     * separate 32 bytes aligned region so that the SAU can programmed to just set
+     * this region as Non-Secure Callable.
+     */
+    .gnu.sgstubs : ALIGN(32)
+    {
+        SG_ALIGN;
+        Load$$LR$$LR_VENEER$$Base = ABSOLUTE(.);
+        *(.gnu.sgstubs*)
+    } > VENEERS AT> VENEERS
+
+    /* This is needed because the linker seems to place the veneers *after* the
+       .gnu.sgstubs output section, so any end alignment applied inside it are
+       placed before the veneers instead of after. */
+    .sg_end_dummy :
+    {
+        SG_END_ALIGN;
+        Load$$LR$$LR_VENEER$$Limit = ABSOLUTE(. - 1);
+    } > VENEERS AT> VENEERS
```

Comment:
But you don't need to define another output section for it. It should be sufficient to just move `SG_END_ALIGN` out of the `.gnu.sgstubs` section.